### DISCUSSION TITLE
GoPro USB WEBCAM DOCKER

### DISCUSSION THREAD

**Discussion by y9-csillagd** [bluenviron/mediamtx#562] [Questions] [answered]
Hi

The program you have written is very good. I'm running it in Docker and I want to stream the image of a USB camera (GoPro), but I don't understand how I can pass the /dev/video42 source to the server. 

Docker Config:
  rtmpServer:
      container_name: rtmpserver
      image: aler9/rtsp-simple-server
      restart: always
      network_mode: default
      ports:
        - "1935:1935"
        - "8888:8888"
        - "8554:8554"
      volumes:
        - /docker/media/rtsp-simple-server.yml:/rtsp-simple-server.yml
      devices:
        - /dev/video42:/dev/video42

Server Config:
paths:
  cam:
    runOnInit: ffmpeg -f v4l2 -i /dev/video42 -f rtmp rtmp://10.10.50.128:1935/live
    runOnInitRestart: yes

Can you help me?
Thanks, David

**Comment by aler9** [comment, marked as answer]
Hello, the Docker image doesn't include ffmpeg, you have to build a Docker image that includes both rtsp-simple-server and ffmpeg.

See here https://github.com/aler9/rtsp-simple-server/issues/183#issuecomment-760856015

**Reply by y9-csillagd** [reply]
Thanks

[a comment by an automated account is not shown]
